### PR TITLE
Search feature

### LINKED ISSUE #50: Search Feature
To implement the following:
- Search /q: finds question with a particular keyword
- Search /a: finds answer with a particular keyword
- Search /t: finds questions & answer set based on the tag
- Search: find keyword in everything

### PR DESCRIPTION
**Pull request by yongmingyang**
Updated the find feature to support:
- find /q (for finding keyword in question)
- find /a (for finding keyword in answer)
- find /t (for finding keyword in tags)
- find {keyword} (for finding keyword in question and/or answer)

Updated UserGuide for find functionality

close #50

### CODE REVIEW

File: src/main/java/seedu/medmoriser/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -27,7 +30,17 @@ public FindCommand parse(String args) throws ParseException {
 
         String[] nameKeywords = trimmedArgs.split("\\s+");
 
-        return new FindCommand(new QuestionContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        String findType = nameKeywords[0];
+        switch (findType) {
+        case "/t":
```
Comment (jonfoocy) [inline src/main/java/seedu/medmoriser/logic/parser/FindCommandParser.java]: Maybe can change to t/ to keep the same format as the add command

File: src/test/java/seedu/medmoriser/logic/parser/FindCommandParserTest.java [hunk 2]
```diff
@@ -9,7 +9,8 @@
 import org.junit.jupiter.api.Test;
 
 import seedu.medmoriser.logic.commands.FindCommand;
-import seedu.medmoriser.model.qanda.QuestionContainsKeywordsPredicate;
+import seedu.medmoriser.model.qanda.QAndAContainsKeywordsPredicate;
+//import seedu.medmoriser.model.qanda.QuestionContainsKeywordsPredicate;
```
Comment (jonfoocy) [inline src/test/java/seedu/medmoriser/logic/parser/FindCommandParserTest.java]: Can delete this import?

**Review by jonfoocy — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/medmoriser/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (joshruien) [reply src/main/java/seedu/medmoriser/logic/parser/FindCommandParser.java]: yes i agree with Jon, this way the user will not be confused.

**Review by joshruien — APPROVED** [review]
Looks good other than the changes suggested by Jon

**Review by jonfoocy — APPROVED** [review]
LGTM

**Pull request merged**